### PR TITLE
update conduit for 0.5.0 release

### PR DESCRIPTION
**Pull request by cyrush**
also a few small updates to the ascent package

### CODE REVIEW

**Comment by cyrush** [comment]
can I get some advice on which old APIs I am using?  (Failed travis ci check?)

**Comment by adamjstewart** [comment]
`setup_environment` was split into `setup_build_environment` and `setup_run_environment`. Same for `setup_dependent_environment`.

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ class Ascent(Package):
     # package dependencies
     ###########################################################################
 
-    depends_on("cmake@3.9.2:3.9.999", type='build')
+    depends_on("cmake@3.14.5", type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]: It doesn't build with newer versions? I think this should be:
```suggestion
    depends_on("cmake@3.14:", type='build')
```

Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: I will relax to allow newer vers

**Comment by cyrush** [comment]
@adamjstewart thank for the help, I switched to use `setup_build_environment`

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ class Ascent(Package):
     #######################
     depends_on("py-sphinx", when="+python+doc", type='build')
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_build_environment(self, spack_env, run_env):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]:
```suggestion
    def setup_build_environment(self, env):
```

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 3]
```diff
@@ -120,7 +120,7 @@ class Ascent(Package):
...
         spack_env.set('CTEST_OUTPUT_ON_FAILURE', '1')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]:
```suggestion
        env.set('CTEST_OUTPUT_ON_FAILURE', '1')
```

**Comment by cyrush** [comment]
@adamjstewart thanks!

**Pull request merged**